### PR TITLE
[extractor/oneplace] Add OnePlacePodcast extractor

### LINKED ISSUE #5543: Support oneplace.com
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a new site support request
- [X] I've verified that I'm running yt-dlp version **2022.11.11** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that none of provided URLs [violate any copyrights](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy) or contain any [DRM](https://en.wikipedia.org/wiki/Digital_rights_management) to the best of my knowledge
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and am willing to share it if required

### Region

United States of America

### Example URLs

Podcast link where you can click on the play icon to play the audio:
https://www.oneplace.com/ministries/a-daily-walk/listen/living-in-the-last-days-part-2-958461.html

Audio Livestream of a talk radio station
https://www.oneplace.com/christian-radio/the-mission-am-980-kkms-133

This link below list links to live audio stations
https://www.oneplace.com/christian-radio/

Playlists and archives:
https://www.oneplace.com/ministries/connect/series/playlist
https://www.oneplace.com/ministries/connect/series/expound-genesis
https://www.oneplace.com/ministries/connect/listen/

### Provide a description that is worded well enough to be understood

oneplace.com is a podcast website. Please support it. Thank you.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
yt-dlp -vU https://www.oneplace.com/ministries/a-daily-walk/listen/living-in-the-last-days-part-2-958461.html
[debug] Command-line config: ['-vU', 'https://www.oneplace.com/ministries/a-daily-walk/listen/living-in-the-last-days-part-2-958461.html']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.11.11 [8b64402] (pip)
[debug] Python 3.10.8 (CPython x86_64 64bit) - Linux-6.0.6-artix1-1-x86_64-with-glibc2.36 (OpenSSL 1.1.1q  5 Jul 2022, glibc 2.36)
[debug] exe versions: ffmpeg 5.1.2 (setts), ffprobe 5.1.2
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-2022.09.24, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1723 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.11.11, Current version: 2022.11.11
yt-dlp is up to date (2022.11.11)
[debug] [generic] Extracting URL: https://www.oneplace.com/ministries/a-daily-walk/listen/living-in-the-last-days-part-2-958461.html
[generic] living-in-the-last-days-part-2-958461: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] living-in-the-last-days-part-2-958461: Extracting information
[debug] Looking for embeds
ERROR: Unsupported URL: https://www.oneplace.com/ministries/a-daily-walk/listen/living-in-the-last-days-part-2-958461.html
Traceback (most recent call last):
  File "/home/user/.local/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 1485, in wrapper
    return func(self, *args, **kwargs)
  File "/home/user/.local/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 1561, in __extract_info
    ie_result = ie.extract(url)
  File "/home/user/.local/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 674, in extract
    ie_result = self._real_extract(url)
  File "/home/user/.local/lib/python3.10/site-packages/yt_dlp/extractor/generic.py", line 2721, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://www.oneplace.com/ministries/a-daily-walk/listen/living-in-the-last-days-part-2-958461.html
```

### PR DESCRIPTION
**Pull request by HobbyistDev**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

</details>

This PR add site support for oneplace.com. This extractor just extract individual audio url (podcast episode)

Partially fixes #5543

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [x] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

File: yt_dlp/extractor/oneplace.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_valid_url(url).group('id')
+        webpage = self._download_webpage(url, video_id)
+
+        media_url = self._search_regex(r'mp3-url="([^"]+)', webpage, 'media_url', fatal=False) or self._search_regex(
+            r'<div[^>]+id\s*=\s*"player"[^>]+data-media-url\s*=\s*"(?P<media_url>[^"]+)', webpage,
+            'media_url')
```
Comment (pukkandan) [inline yt_dlp/extractor/oneplace.py:23]:
```suggestion
        media_url = self._search_regex((
            r'mp3-url\s*=\s*"([^"]+)',
            r'<div[^>]+id\s*=\s*"player"[^>]+data-media-url\s*=\s*"(?P<media_url>[^"]+)',
        ), webpage, 'media url')
```

File: yt_dlp/extractor/oneplace.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+        return {
+            'id': video_id,
+            'url': media_url,
+            'title': clean_html(self._search_regex(
```
Comment (pukkandan) [inline yt_dlp/extractor/oneplace.py:28]: Why not `_search_html_regex`

File: yt_dlp/extractor/oneplace.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+                r'<div[^>]class\s*=\s*"details"[^>]+>[^<]<h2[^>]+>(?P<title>[^>]+)>', webpage,
+                'title', fatal=False, default=None)) or self._html_search_meta(['og:title', 'title'], webpage),
+            'ext': 'mp3',
```
Comment (pukkandan) [inline yt_dlp/extractor/oneplace.py:31]:
```suggestion
            'ext': 'mp3',
            'vcodec': 'none',
```

File: yt_dlp/extractor/oneplace.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+            'description': clean_html(self._search_regex(
```
Comment (pukkandan) [inline yt_dlp/extractor/oneplace.py:32]: Why not _search_html_regex

File: yt_dlp/extractor/oneplace.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+                r'<div[^>]+class="[^"]+epDesc"[^>]*>\s*(?P<desc>.+)', webpage, 'description',
+                fatal=False, default=None))
```
Comment (pukkandan) [inline yt_dlp/extractor/oneplace.py:34]:
```suggestion
                r'<div[^>]+class="[^"]+epDesc"[^>]*>\s*(?P<desc>.+)', webpage, 'description', default=None))
```
default implies non-fatal

File: yt_dlp/extractor/oneplace.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+        return {
+            'id': video_id,
+            'url': media_url,
+            'title': clean_html(self._search_regex(
+                r'<div[^>]class\s*=\s*"details"[^>]+>[^<]<h2[^>]+>(?P<title>[^>]+)>', webpage,
+                'title', fatal=False, default=None)) or self._html_search_meta(['og:title', 'title'], webpage),
+            'ext': 'mp3',
+            'description': clean_html(self._search_regex(
+                r'<div[^>]+class="[^"]+epDesc"[^>]*>\s*(?P<desc>.+)', webpage, 'description',
```
Comment (pukkandan) [inline yt_dlp/extractor/oneplace.py:33]: .+ may capture too much

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/oneplace.py (hunk 6 above)
Comment (pukkandan) [reply yt_dlp/extractor/oneplace.py]: anything we can do about this?

Comment (HobbyistDev) [reply yt_dlp/extractor/oneplace.py]: How about bd8ecc28d7560b6a0735712a1bb4cfe6bf402150 ? not a proper solution though.

Comment (Lesmiscore) [reply yt_dlp/extractor/oneplace.py]: a little twist: `(?P<desc>.+?)</div>` (unless we have to assume nested <div>)

**Pull request merged**